lib/libc/common/source/stdlib/malloc.c

```diff
@@ -114,7 +114,7 @@ extern char _heap_sentry[];
 
 # endif /* else ALLOCATE_HEAP_AT_STARTUP */
 
-Z_LIBC_DATA static struct sys_heap z_malloc_heap;
+Z_LIBC_DATA struct sys_heap z_malloc_heap;
```

Comment:
I understand this can be useful for debugging purposes; but, this more or less creates an implicit contract between an internal implementation detail of common libc and whatever will be using this (another Zephyr component? user application?), and I anticipate we will be seeing bugs like "unresolved symbol `z_malloc_heap`" (or even worse "`sys_heap_runtime_stats_get` crashing because `z_malloc_heap` is no longer `struct sys_heap`") in the future if we do this.

I would say we should define and implement Zephyr-flavoured `mallinfo` with the fields we need if we want to make the internal heap stats available to the components external to the common libc; otherwise, patches/hacks like these should stay in one's local branch.